enhancements/sig-architecture/224-event-based-manifestwork/README.md

```diff
@@ -0,0 +1,360 @@
+
+To ensure uniqueness and immutability of source components, each source, whether it is a consumer on the open cluster management hub cluster or a restful service handling resource requests, should possess a unique identifier. For instance, the open cluster management hub consumer can generate a source ID by hashing the hub cluster URL and appending a consumer name. Similarly, a restful service can choose a unique name or generate a unique ID in the associated database for its source identification.
+
+### Spec Event
+
+Sources send their resources along with the message to the work agent, enabling the creation, updating, or deletion of the respective resources on the managed cluster.
+
+Event transfer flow:
+
+1. The work agent subscribes to the spec event topic `/sources/+/clusters/<cluster-id>/manifests` from the broker, for example: `/sources/+/clusters/cluster1/manifests`.
+2. The source receives high level API request and publishes its resources using the spec event to the topic `/sources/<source-id>/clusters/<cluster-id>/manifests`, for example: `/sources/sd3ded4-mwrs-controller/clusters/cluster1/manifests`.
+3. The work agent receives the published event from the source and deploys the manifests included in the event.
+
+```mermaid
+sequenceDiagram
+    WorkAgent -->> Broker: Subscribe topic - /sources/+/clusters/<cluster-id>/manifets
+    Source -->> Source: Receive high-level API request and encode manifestwork into spec event
+    Source -->> Broker: Publish spec event for manifestwork
+    Broker -->> WorkAgent: Send spec event for manifestwork
+    WorkAgent -->> WorkAgent: Decode spec event and apply manifestwork
+```
+
+Event Schema:
+
+```json
+{
+    "specversion": "1.0", //The version of the CloudEvents specification which the event uses.
+    "id": "<cloud-events-message-id>",
+    "source": "<source-id>", //Identifies the context in which an event happened. Often this will include information such as the type of the event source, the organization publishing the event or the process that produced the event.
+    "type": "io.open-cluster-management.v1.manifestwork.spec.created", //This attribute contains a value describing the type of event related to the originating occurrence. Often this attribute is used for routing, observability, policy enforcement, etc.
+    "time": "2023-04-05T17:31:00Z", // Timestamp of when the occurrence happened.
+    "datacontenttype": "application/json",
+    // resourceID is the unique id for this manifestwork.
+    // It is represented in string.
+    // It is a required property.
+    "resourceID": "748d04ad-887a-44dd-b788-da1373227072", 
+    // resourceVersion is a sequence number that grows as the data of this message changes.
+    // It is represented in int64.
+    // It is a required property.
```

Comment:
let's clarify that it is source's responsibility to ensure it is incremental

Reply:
done.